tests/codegen/unwind-landingpad-inline.rs

```diff
@@ -0,0 +1,37 @@
+// min-llvm-version: 15.0.0
+// compile-flags: -Copt-level=3
+#![crate_type = "lib"]
+
+// This test checks that we can inline drop_in_place in
+// unwind landing pads.
+
+// Without inlining, the box pointers escape via the call to drop_in_place,
+// and LLVM will not optimize out the pointer comparison.
+// With inlining, everything should be optimized out.
+// See https://github.com/rust-lang/rust/issues/46515
+// CHECK-LABEL: @check_no_escape_in_landingpad
+// CHECK: start:
+// CHECK-NEXT: ret void
+#[no_mangle]
+pub fn check_no_escape_in_landingpad(f: fn()) {
+    let x = &*Box::new(0);
+    let y = &*Box::new(0);
+
+    if x as *const _ == y as *const _ {
+        f();
+    }
+}
+
+// Without inlining, the compiler can't tell that
+// dropping an empty string (in a landing pad) does nothing.
+// With inlining, the landing pad should be optimized out.
+// See https://github.com/rust-lang/rust/issues/87055
+// CHECK-LABEL: @check_eliminate_noop_drop
+// CHECK: start:
+// CHECK-NEXT: call void %g()
```

Comment:
Okay, it looks like this indeed doesn't optimize as expected. From the IR dump, it should be eligible for inlining, but it still doesn't happen for some reason. We should use `CHECK-NOT: call void %g()` here and add a TODO, and not mark that issue as as fixed.

Reply:
Is there any way I can run the CI enviroment locally? This is what my `unwind-landingpad-inline.ll` file looks like:
```
; ModuleID = 'unwind_landingpad_inline.8bb05d3a-cgu.0'
source_filename = "unwind_landingpad_inline.8bb05d3a-cgu.0"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"

; Function Attrs: nonlazybind uwtable
define void @check_no_escape_in_landingpad(ptr nocapture noundef nonnull readonly %f) unnamed_addr #0 personality ptr @rust_eh_personality {
start:
  ret void
}

; Function Attrs: nonlazybind uwtable
define void @check_eliminate_noop_drop(ptr nocapture noundef nonnull readonly %g) unnamed_addr #0 personality ptr @rust_eh_personality {
start:
  tail call void %g()
  ret void
}

; Function Attrs: nonlazybind uwtable
declare noundef i32 @rust_eh_personality(i32, i32 noundef, i64, ptr, ptr) unnamed_addr #0

attributes #0 = { nonlazybind uwtable "probe-stack"="__rust_probestack" "target-cpu"="x86-64" }

!llvm.module.flags = !{!0, !1}

!0 = !{i32 7, !"PIC Level", i32 2}
!1 = !{i32 2, !"RtLibUseGOT", i32 1}
```

Reply:
`src/ci/docker/run.sh x86_64-gnu-stable` should reproduce the CI environment.

Sometimes codegen tests are affected by libstd being compiled with debug assertions, which I believe is also the case for `x86_64-gnu-stable`. If that's the issue, adding `ignore-debug` should fix it.

Reply:
Hmm, I'm getting
```
+ python3 ../x.py --stage 2 test
Building rustbuild
    Finished dev [unoptimized] target(s) in 0.02s
error: could not find commit hash for downloading LLVM
help: maybe your repository history is too shallow?
help: consider disabling `download-ci-llvm`
help: or fetch enough history to include one upstream commit
thread 'main' panicked at 'explicit panic', native.rs:166:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Build completed unsuccessfully in 0:00:00
== clock drift check ==
  local time: Sat Feb  4 18:07:25 UTC 2023
  network time: Sat, 04 Feb 2023 18:07:25 GMT
== end clock drift check ==
```
when I try to run ` sh src/ci/docker/run.sh x86_64-gnu` (stable cannot be found).